src/Propel/Runtime/ActiveQuery/InstancePoolTrait.php

```diff
@@ -30,8 +30,9 @@ public static function addInstanceToPool($object, $key = null)
             if ($key === null) {
                 $key = static::getInstanceKey($object);
             }
-
-            self::$instances[$key] = $object;
+            if (!empty($key)) {
```

Comment:
So this is an empty string here then? as we do check for null before.
I wonder if the code calling this would check for proper valid "non empty string" instead, and doesnt call this method then in the first place?

Making this a no-op method in such an edge case might make it more difficult to debug in the future.

Reply:
I think it is the simplest solution, other solutions will involve adding more complexity to the development for the rare case that a table does not have a primary key.